internal/uploader/events_uploader_test.go

```diff
@@ -0,0 +1,299 @@
+		buf, formType, err := uploader.prepareBody(ctx, &clusterID, mockEvents)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(buf.Bytes()).NotTo(BeNil())
+		Expect(formType).NotTo(BeEmpty())
+	})
+	It("fails to create the request body when there's no data", func() {
+		mockEvents.EXPECT().V2GetEvents(ctx, &clusterID, nil, nil, models.EventCategoryMetrics, models.EventCategoryUser).Return(
+			nil, errors.New("no events found")).Times(1)
+		buf, formType, err := uploader.prepareBody(ctx, &clusterID, mockEvents)
+		Expect(err).To(HaveOccurred())
+		Expect(buf.Bytes()).To(BeNil())
+		Expect(formType).To(BeEmpty())
+	})
+})
+
+var _ = Describe("prepareFiles", func() {
+	var (
+		ctrl       *gomock.Controller
+		db         *gorm.DB
+		dbName     string
+		clusterID  strfmt.UUID
+		mockEvents *eventsapi.MockHandler
+		ctx        context.Context
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		db, dbName = common.PrepareTestDB()
+		clusterID = strfmt.UUID(uuid.New().String())
+		ctx = context.Background()
+		mockEvents = eventsapi.NewMockHandler(ctrl)
+	})
+
+	AfterEach(func() {
+		common.DeleteTestDB(db, dbName)
+	})
+
+	It("successfully prepares all files", func() {
+		mockEvents.EXPECT().V2GetEvents(
+			ctx, &clusterID, nil, nil, models.EventCategoryMetrics, models.EventCategoryUser).Return([]*common.Event{}, nil).Times(1)
+
+		infraEnvID := strfmt.UUID(uuid.New().String())
+		hostID := strfmt.UUID(uuid.New().String())
+		createTestObjects(db, &clusterID, &hostID, &infraEnvID)
+
+		buf := new(bytes.Buffer)
+		err := prepareFiles(ctx, db, &clusterID, mockEvents, buf)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(buf.Bytes()).NotTo(BeEmpty())
+
+		gzipReader, err := gzip.NewReader(bytes.NewReader(buf.Bytes()))
+		tarReader := tar.NewReader(gzipReader)
+		for {
+			hdr, err := tarReader.Next()
+			if err == io.EOF {
+				break
+			}
+			Expect(err).NotTo(HaveOccurred())
+			switch hdr.Name {
+			case fmt.Sprintf("%s/events/hosts.json", clusterID):
+				cluster, err := common.GetClusterFromDBWithHosts(db, clusterID)
+				Expect(err).NotTo(HaveOccurred())
+				hostsJson, err := json.MarshalIndent(cluster.Cluster.Hosts, "", " ")
+				Expect(err).NotTo(HaveOccurred())
+				checkFileContents(tarReader, hostsJson)
+			case fmt.Sprintf("%s/events/infraenv.json", clusterID):
+				infraEnv, err := common.GetInfraEnvFromDB(db, infraEnvID)
+				Expect(err).NotTo(HaveOccurred())
+				infraEnvJson, err := json.Marshal(infraEnv.InfraEnv)
+				Expect(err).NotTo(HaveOccurred())
+				checkFileContents(tarReader, infraEnvJson)
+			case fmt.Sprintf("%s/events/cluster.json", clusterID):
+				cluster, err := common.GetClusterFromDBWithHosts(db, clusterID)
+				Expect(err).NotTo(HaveOccurred())
+				clusterJson, err := json.Marshal(cluster.Cluster)
+				Expect(err).NotTo(HaveOccurred())
+				checkFileContents(tarReader, clusterJson)
+			case fmt.Sprintf("%s/events.json", clusterID):
+				eventJson, err := json.MarshalIndent([]*common.Event{}, "", " ")
```

Comment:
Should we be testing with some events?

Reply:
Good idea! I've tried to do it [here](https://github.com/openshift/assisted-service/pull/4880/commits/248f4ac043767feb8bbc6843cd7f1e09c143f509#diff-7eb6290c1b1e1c5de47c84bd0946399c43bb374390d36e89483bfb0a5af7e122R220-R237) but I'm not sure it's a good test 😅 

Are there better tests to do or should I be testing in a different place?